src/textual/dom.py

```diff
@@ -1276,17 +1275,27 @@ def toggle_class(self, *class_names: str) -> Self:
         self._update_styles()
         return self
 
-    def has_pseudo_class(self, *class_names: str) -> bool:
-        """Check for pseudo classes (such as hover, focus etc)
+    def has_pseudo_class(self, class_name: str) -> bool:
```

Comment:
In light of the optimisation intent, isn't it faster to check specifically for the given pseudo class instead of iterating over the generator of all pseudo classes?

Reply:
At the moment, the iterator is the only way we have of exposing the pseudo classes. Its only actually tests which use `has_pseudo_class` currently.

Reply:
I don't the understand the second sentence _at all_... 🙃

Reply:
Search the code for "has_pseudo_class". It is only used by tests.

Reply:
Understood.